ProfileManager.js

```diff
@@ -938,9 +949,18 @@ export default class ProfileManager {
   async _getAgentSigner({profileAgentId, useEphemeralSigner}) {
     const cacheKey = `${profileAgentId}-${useEphemeralSigner}`;
     const agentSignersCache = this._getCache('agent-signers');
-    const agentSigner = agentSignersCache.get(cacheKey);
+    const agentSigner = await agentSignersCache.get(cacheKey);
     if(agentSigner) {
-      return agentSigner;
+      const capability = agentSigner.capability;
+      if(capability && capability.expires) {
+        const now = Date.now();
+        const expiryDate = new Date(capability.expires);
+        if(expiryDate.getTime() > now) {
+          return agentSigner;
+        }
+      } else {
+        return agentSigner;
+      }
```

Comment:
If you reverse the conditionals, it's a bit cleaner because you can have an early return:

```suggestion
      const capability = agentSigner.capability;
      if(!(capability && capability.expires)) {
        return agentSigner;
      }
      const now = Date.now();
      const expiryDate = new Date(capability.expires);
      if(expiryDate.getTime() > now) {
        return agentSigner;
      }
      agentSignersCache.del(cacheKey);
```

But, just like above, the zcap has expired, I think we want to delete the entry from the cache so it will get refreshed. Also, we'll want to add that grace period thing in here as well.